### PR TITLE
[dhcp_server_test] Fix log errors caused by dhcp server tests

### PR DESCRIPTION
**Pull request by w1nda**
### Description of PR

Summary:
Fixes # (issue)
When updating config db, there are some invalid references modeled by yang model.

For example: ERR sonic_yang: Data Loading Failed:Leafref "/sonic-dhcp-server-ipv4:sonic-dhcp-server-ipv4/sonic-dhcp-server-ipv4:DHCP_SERVER_IPV4_RANGE/sonic-dhcp-server-ipv4:DHCP_SERVER_IPV4_RANGE_LIST/sonic-dhcp-server-ipv4:name" of value "range_192.168.0.91" points to a non-existing leaf.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [x] 202311
- [x] 202405

#### What is the motivation for this PR?
When updating config db, there are some invalid references modeled by yang model.

we can adjust the order of applying patches to avoid invalid status.

#### How did you do it?
Adjust the order of config patches

#### How did you verify/test it?
Run tests

### CODE REVIEW

**Comment by wsycqyz** [comment]
Re-trigger PR test.

File: tests/dhcp_server/test_dhcp_server_multi_vlans.py [hunk 1]
```diff
@@ -250,6 +237,7 @@ def test_single_ip_assignment(
             server_id=gateway,
             net_mask=net_mask
         )
+    clean_dhcp_server_config(duthost)
```
Comment (yaqiangz) [inline tests/dhcp_server/test_dhcp_server_multi_vlans.py:240]: Would it be better to clean by fixcture?

**Review by yaqiangz — APPROVED** [review]

File: tests/dhcp_server/test_dhcp_server_multi_vlans.py (hunk 1 above)
Comment (w1nda) [reply tests/dhcp_server/test_dhcp_server_multi_vlans.py]: Yes, I moved the common config clean to new fixture clean_dhcp_server_after_test

**Review by yejianquan — APPROVED** [review]
LGTM

**Pull request merged**